atlasdb-perf/src/main/java/com/palantir/atlasdb/performance/benchmarks/KvsGetRangeBenchmarks.java

```diff
@@ -0,0 +1,196 @@
+import org.openjdk.jmh.annotations.Benchmark;
+import org.openjdk.jmh.annotations.BenchmarkMode;
+import org.openjdk.jmh.annotations.Level;
+import org.openjdk.jmh.annotations.Measurement;
+import org.openjdk.jmh.annotations.Mode;
+import org.openjdk.jmh.annotations.OutputTimeUnit;
+import org.openjdk.jmh.annotations.Scope;
+import org.openjdk.jmh.annotations.Setup;
+import org.openjdk.jmh.annotations.State;
+import org.openjdk.jmh.annotations.TearDown;
+import org.openjdk.jmh.annotations.Warmup;
+
+import com.google.common.collect.Iterables;
+import com.google.common.collect.Lists;
+import com.google.common.collect.Maps;
+import com.google.common.collect.Sets;
+import com.google.common.primitives.Ints;
+import com.palantir.atlasdb.keyvalue.api.Cell;
+import com.palantir.atlasdb.keyvalue.api.KeyValueService;
+import com.palantir.atlasdb.keyvalue.api.RangeRequest;
+import com.palantir.atlasdb.keyvalue.api.RowResult;
+import com.palantir.atlasdb.keyvalue.api.TableReference;
+import com.palantir.atlasdb.keyvalue.api.Value;
+import com.palantir.atlasdb.performance.backend.KeyValueServiceConnector;
+import com.palantir.common.base.ClosableIterator;
+import com.palantir.util.paging.TokenBackedBasicResultsPage;
+
+@State(Scope.Benchmark)
+@BenchmarkMode(Mode.SampleTime)
+@OutputTimeUnit(TimeUnit.MICROSECONDS)
+@Warmup(iterations = 1, time = 5, timeUnit = TimeUnit.SECONDS)
+@Measurement(iterations = 1, time = 30, timeUnit = TimeUnit.SECONDS)
+public class KvsGetRangeBenchmarks {
+
+    private static final String TABLE_NAME_1 = "performance.table1";
+    private static final String ROW_COMPONENT = "key";
+    private static final String COLUMN_NAME = "value";
+    private static final byte [] COLUMN_NAME_IN_BYTES = COLUMN_NAME.getBytes();
+    private static final long DUMMY_TIMESTAMP = 1L;
+    private static final long QUERY_TIMESTAMP = 2L;
+
+    private static final int VALUE_BYTE_ARRAY_SIZE = 100;
+    private static final long VALUE_SEED = 279L;
+
+    private KeyValueServiceConnector connector;
+    private KeyValueService kvs;
+    private Random random = new Random(VALUE_SEED);
+
+    private TableReference tableRef1;
+
+    private static final int NUM_ROWS = 10000;
+    private static final int PUT_BATCH_SIZE = 1000;
+    private static final int NUM_REQUESTS = 1000;
+
+    /**
+     * Should throw on failure.
+     */
+    private Runnable verifier;
+
+    private static Runnable NO_OP = new Runnable() {
+        @Override
+        public void run() {
+            return;
+        }
+    };
+
+
+    @Setup(Level.Trial)
+    public void setup(KeyValueServiceConnector connector) {
+        this.connector = connector;
+        this.kvs = connector.connect();
+        this.tableRef1 = KvsBenchmarks.createTable(kvs, TABLE_NAME_1, ROW_COMPONENT, COLUMN_NAME);
+        storeData();
+    }
+
+    @Setup(Level.Invocation)
+    public void setupInvocation() {
+        verifier = NO_OP;
+    }
+
+    @TearDown(Level.Invocation)
```

Comment:
Invocation level setup/teardown is quite expensive, and not sure you really want to do this see http://hg.openjdk.java.net/code-tools/jmh/file/tip/jmh-core/src/main/java/org/openjdk/jmh/annotations/Level.java#l50